### ISSUE TITLE
Only include `[dir=rtl]` code if a settings variable has been set for an rtl mode?

### ISSUE THREAD

**Issue by sean-dunwoody** [studio24/amplify#12]
There is a significant amount of code in the default Amplify build that accounts for text that can be read right to left (in the `core.css compiled file, searching for `[dir=rtl]` returns 38 results).

This is helpful to have if needed. But to date, I can only think of one project (W3C) where this has been necessary.

It's entirely possible that I have forgotten one or two other projects where we have needed to accommodate rtl text, but it is generally a very rare occurrence.

As such, for performance reasons, it might be good if we were to only output the right-to-left code if a variable is set in the Amplify config to allow for this.

For example we could change code like this:

```scss
[dir="rtl"] {
	summary {
		padding-left: 0;
		padding-right: rem(25);

		&::before {
			left: auto;
			right: 0;
			transform: rotate(180deg);
		}
	}
}
```

To this:

```scss
// this variable would be set in a central settings file ideally
$enable-rtl: true;

@if $enable-rtl {
	[dir="rtl"] {
		summary {
			padding-left: 0;
			padding-right: rem(25);

			&::before {
				left: auto;
				right: 0;
				transform: rotate(180deg);
			}
		}
	}
}
```

(utilising the if statement provided by sass: [https://sass-lang.com/documentation/at-rules/control/if](https://sass-lang.com/documentation/at-rules/control/if)) 

Advantages:

1. Smaller code output for sites that don't need rtl styling (the alternative is to go through and remove each instance, which is time consuming, and maybe problematic if rtl support needs adding to the project at a later date). This would offer a modest performance improvement to the end user.
2. There might be other changes that we want to make based on the `$enable-rtl` value? Although I think this is probably unlikely

Disadvantages:

1. Might be frustrating if you end up working on a project where right-to-left support is a necessity; as you would either need to maintain this check for all new code, or remove any references to it (which could take a while). 
2. Adding an extra variable, and a small amount of complexity, to the scss files in the project

**Comment by NicolaSaunders** [comment]
@sean-dunwoody You are absolutely right about there being a lot of extra code, and it has indeed come from the W3C project. I pondered whether to leave it in or remove it, and in the end I left it.

There is an even simpler alternative to your suggestion, which is to replace traditional directional styles with logical properties. If we are ready to deprioritise IE11, I think this would be the best route to take - they have [pretty good browser support](https://caniuse.com/?search=logical%20properties).

What do we think @sean-dunwoody @maries24 @simonrjones ??

**Comment by simonrjones** [comment]
@NicolaSaunders are you saying if we use logical properties this should work with ltr both and rtl writing directions? If so, that does sound a good way forward. How much work do you think this is likely to take? And do you think this is worth doing? 

We need to decide on what min browser support we're expecting for Amplify now, since this has an impact on this.

**Comment by NicolaSaunders** [comment]
Yes, that is what I'm saying. For things like margin and padding, which seem to be well supported as logical properties, they will automatically work with both ltr and rtl writing directions. Some other logical properties don't have as great a browser support, and they are not supported at all in IE11 and non-chromium Edge.

As ever, it all boils down to what we want our standard browser support to be!

**Comment by NicolaSaunders** [comment]
I am now closing this issue as it has been addressed with the release of [version 2.0.0](https://github.com/studio24/amplify/releases/tag/v2.0.0), which introduces logical properties.
